### PR TITLE
Trollheim and Weiss Region overhaul

### LINKED ISSUE #259: Trollweiss Salt Mines have undefined objects
Trollweiss Salt mines does not have any textures on rocks or walls, area is defined a black; should likely be a cold blue/white like other snowy areas.

But report is for a reminder.

Reported by:
https://discord.com/channels/886733267284398130/886738668109307935/1163155595373858958

### PR DESCRIPTION
**Pull request by SirFancyBacon**
Overhauls the Trollheim and Weiss region:
* Remove Normals from vertical dirt to avoid ugly Normals around Trollheim
* Retextures Trollheim with a different Grunge material
* Apply materials to objects missing materials such as rocks, detritus, doors, ladders, walls coffins, etc
* Apply Winter Cobblestone for cobble in Weiss
* Newly generated normals and improved displacement map for Cobblestone
* Replace all snow textures with matching source ones from AmbientCG

image
Fixes #259

### CODE REVIEW

**Review by aHooder — CHANGES_REQUESTED** [review]
I think this snow looks too patchy with those dark spots, and the normals could do with more finer detail ideally:
image

I didn't notice the jagged stone displacement at first, but it looks pretty nice and subtle :slightly_smiling_face: 

The entrance light looks a bit too blown out, but I do like the look a lot otherwise: 
image

I like pretty much all of the other lighting you've added :slightly_smiling_face: 

I unfortunately can't look through any of the stuff in the quest area. Did you change anything there?

File: src/main/java/rs117/hd/data/materials/Material.java [hunk 1]
```diff
@@ -335,6 +341,9 @@ public enum Material {
 	),
 	GRUNGE_1,
 	GRUNGE_2,
+	GRUNGE_2_EADGARS_CAVE_FIX(GRUNGE_2, p -> p.setBrightness(0.65f)),
+	GRUNGE_2_TROLLHEIM_WALL_FIX_1(GRUNGE_2, p -> p.setBrightness(1.8f)),
+	GRUNGE_2_TROLLHEIM_WALL_FIX_2(GRUNGE_2, p -> p.setBrightness(1.2f)),
```
[the remark this answers is not in the document]
Comment (SirFancyBacon) [reply src/main/java/rs117/hd/data/materials/Material.java:346]: Absolutely that would be fantastic! though if we could simply stop whatever is causing the darkness of the object that would be awesome too; Sometimes it is the model; But it's almost like Jagex used dynamic baked lighting on the models in the Troll stronghold; Same model but different levels of darkness

**Comment by SirFancyBacon** [comment]
> I think this snow looks too patchy with those dark spots, and the normals could do with more finer detail ideally: image
> 
> I didn't notice the jagged stone displacement at first, but it looks pretty nice and subtle 🙂
> 
> The entrance light looks a bit too blown out, but I do like the look a lot otherwise: image
> 
> I like pretty much all of the other lighting you've added 🙂
> 
> I unfortunately can't look through any of the stuff in the quest area. Did you change anything there?

I agree, that area is a bit blown out; I think i know how i can fix it too.

I changed EVERYTHING; Quest area too.

File: src/main/java/rs117/hd/data/materials/Material.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (aHooder) [reply src/main/java/rs117/hd/data/materials/Material.java:346]: Presumably Jagex just has different colored variants of it, but I'm not sure. Most likely we can't get a hold of that kind of info, unfortunately. I think we could add some sort of configurable lightness multiplier though

**Comment by SirFancyBacon** [comment]
image

**Comment by aHooder** [comment]
That looks better. Could it maybe make sense to move the light a bit closer to the bottom of the stairs, and making the radius even wider also?

**Comment by SirFancyBacon** [comment]
Snow textures and normals improved so they tile better
image

image

**Comment by SirFancyBacon** [comment]
> That looks better. Could it maybe make sense to move the light a bit closer to the bottom of the stairs, and making the radius even wider also?

image

**Review by aHooder — APPROVED** [review]

**Pull request merged**